Src/Fido2.Models/CredentialCreateOptions.cs

```diff
@@ -55,6 +55,52 @@ public sealed class CredentialCreateOptions : Fido2ResponseBase
     [JsonPropertyName("authenticatorSelection")]
     public AuthenticatorSelection AuthenticatorSelection { get; set; }
 
+    private IReadOnlyList<PublicKeyCredentialHint> _hints = Array.Empty<PublicKeyCredentialHint>();
+
+    /// <summary>
+    /// Guides the user agent in interacting with the user. This OPTIONAL member contains zero or more elements from <see cref="PublicKeyCredentialHint" />.
+    /// </summary>
+    /// <remarks>
+    /// When <see cref="Hints"/> is set, <see cref="AuthenticatorSelection.AuthenticatorAttachment" /> will be set to one of the values in the table below:
+    /// <list type="table">
+    /// <item><term><see cref="PublicKeyCredentialHint.SecurityKey"/></term><description><see cref="AuthenticatorAttachment.CrossPlatform"/></description></item>
+    /// <item><term><see cref="PublicKeyCredentialHint.ClientDevice"/></term><description><see cref="AuthenticatorAttachment.Platform"/></description></item>
+    /// <item><term><see cref="PublicKeyCredentialHint.Hybrid"/></term><description><see cref="AuthenticatorAttachment.CrossPlatform"/></description></item>
+    /// </list>
+    /// </remarks>
+    [JsonPropertyName("hints")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public IReadOnlyList<PublicKeyCredentialHint> Hints
+    {
+        get
+        {
+            return _hints;
+        }
+        set
+        {
+            if (value.Any())
+            {
+                switch (value[0])
+                {
+                    case PublicKeyCredentialHint.SecurityKey:
+                    case PublicKeyCredentialHint.Hybrid:
+                        {
+                            AuthenticatorSelection.AuthenticatorAttachment ??= new AuthenticatorAttachment();
+                            AuthenticatorSelection.AuthenticatorAttachment = AuthenticatorAttachment.CrossPlatform;
```

Comment:
@jonashendrickx Maybe I'm missing something, but what's the reason for double assignment to `AuthenticatorSelection.AuthenticatorAttachment`?

Reply:
@abergs It had to be `AuthenticatorSelection` which I had to initialize, as it may be null.

> authenticatorSelection, of type [AuthenticatorSelectionCriteria](https://www.w3.org/TR/webauthn-3/#dictdef-authenticatorselectioncriteria)
> The [Relying Party](https://www.w3.org/TR/webauthn-3/#relying-party) MAY use this OPTIONAL member to specify capabilities and settings that the [authenticator](https://www.w3.org/TR/webauthn-3/#authenticator) MUST or SHOULD satisfy to participate in the [create()](https://w3c.github.io/webappsec-credential-management/#dom-credentialscontainer-create) operation. See [§ 5.4.4 Authenticator Selection Criteria (dictionary AuthenticatorSelectionCriteria)](https://www.w3.org/TR/webauthn-3/#dictionary-authenticatorSelection).